src/MaIN.Domain/Entities/Agents/AgentSource/AgentFileSourceDetails.cs

```diff
@@ -2,7 +2,6 @@ namespace MaIN.Domain.Entities.Agents.AgentSource;
 
 public class AgentFileSourceDetails : AgentSourceDetailsBase, IAgentSource
 {
-    public required string Path { get; init; }
-    public required string Name { get; init; }
+    public required Dictionary<string, string> Files { get; init; } = new();
```

Comment:
I like approach taken, however for simplicity measures it would be nice I think to allow providing list of files (paths) only, then in FechCommand/Services logic you can take file name from the path with Path class methods. (for example https://learn.microsoft.com/pl-pl/dotnet/api/system.io.path.getfilename?view=net-8.0 )

Ideally would be nice to apply both cases, so user can specify name, and otherwise its taken from path